network/ipvlan.md

```diff
@@ -0,0 +1,629 @@
+`--internal` flag is used, a netlink type `dummy` parent interface is created
+for the user and used as the parent interface effectively isolating the network
+completely.
+
+The following two `docker network create` examples result in identical networks
+that you can attach container to:
+
+```console
+# Empty '-o parent=' creates an isolated network
+$ docker network  create -d ipvlan \
+    --subnet=192.168.10.0/24 isolated1
+
+# Explicit '--internal' flag is the same:
+$ docker network  create -d ipvlan \
+    --subnet=192.168.11.0/24 --internal isolated2
+
+# Even the '--subnet=' can be left empty and the default 
+# IPAM subnet of 172.18.0.0/16 will be assigned
+$ docker network  create -d ipvlan isolated3
+
+$ docker run --net=isolated1 --name=cid1 -it --rm alpine /bin/sh
+$ docker run --net=isolated2 --name=cid2 -it --rm alpine /bin/sh
+$ docker run --net=isolated3 --name=cid3 -it --rm alpine /bin/sh
+
+# To attach to any use `docker exec` and start a shell
+$ docker exec -it cid1 /bin/sh
+$ docker exec -it cid2 /bin/sh
+$ docker exec -it cid3 /bin/sh
+```
+
+## IPvlan 802.1q trunk L2 mode example usage
+
+Architecturally, IPvlan L2 mode trunking is the same as Macvlan with regard to
+gateways and L2 path isolation. There are nuances that can be advantageous for
+CAM table pressure in ToR switches, one MAC per port and MAC exhaustion on a
+host's parent NIC to name a few. The 802.1q trunk scenario looks the same. Both
+modes adhere to tagging standards and have seamless integration with the physical
+network for underlay integration and hardware vendor plugin integrations.
+
+Hosts on the same VLAN are typically on the same subnet and almost always are
+grouped together based on their security policy. In most scenarios, a multi-tier
+application is tiered into different subnets because the security profile of each
+process requires some form of isolation. For example, hosting your credit card
+processing on the same virtual network as the frontend webserver would be a
+regulatory compliance issue, along with circumventing the long standing best
+practice of layered defense in depth architectures. VLANs or the equivocal VNI
+(Virtual Network Identifier) when using the Overlay driver, are the first step
+in isolating tenant traffic.
+
+![Docker VLANs in Depth](images/vlans-deeper-look.png)
+
+The Linux sub-interface tagged with a vlan can either already exist or will be
+created when you call a `docker network create`. `docker network rm` will delete
+the sub-interface. Parent interfaces such as `eth0` are not deleted, only
+sub-interfaces with a netlink parent index > 0.
+
+For the driver to add/delete the vlan sub-interfaces the format needs to be
+`interface_name.vlan_tag`. Other sub-interface naming can be used as the
+specified parent, but the link will not be deleted automatically when
+`docker network rm` is invoked. 
+
+The option to use either existing parent vlan sub-interfaces or let Docker manage
```

Comment:
We've used both `vlan` and `VLAN` in this doc. Suggest we use VLAN throughout for consistency,